Framework/Kernel/inc/MantidKernel/PropertyWithValue.h

```diff
@@ -637,6 +641,29 @@ template <typename TYPE> class DLLExport PropertyWithValue : public Property {
   PropertyWithValue();
 };
 
+#define PROPERTYWITHVALUE_SAVEPROPERTY(type)                                   \
+  template <>                                                                  \
+  inline void PropertyWithValue<type>::saveProperty(::NeXus::File *file) {     \
+    file->makeGroup(this->name(), "NXlog", 1);                                 \
+    file->writeData("value", m_value);                                         \
+    file->closeGroup();                                                        \
+  }
+
+PROPERTYWITHVALUE_SAVEPROPERTY(float)
+PROPERTYWITHVALUE_SAVEPROPERTY(double)
+PROPERTYWITHVALUE_SAVEPROPERTY(int32_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(uint32_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(int64_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(uint64_t)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::string)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::vector<double>)
+PROPERTYWITHVALUE_SAVEPROPERTY(std::vector<int32_t>)
+
+template <typename TYPE>
+inline void PropertyWithValue<TYPE>::saveProperty(::NeXus::File *file) {
+  this->saveProperty(file);
+}
```

Comment:
I marked them inline to prevent an ODR violation. Is this incorrect? 

Moving `saveProperty` to the implementation file generates several linking errors (see below). 

```
Undefined symbols for architecture x86_64:
  "Mantid::Kernel::PropertyWithValue<boost::shared_ptr<Mantid::Kernel::PropertyManager> >::saveProperty(NeXus::File*)", referenced from:
      vtable for Mantid::Kernel::PropertyWithValue<boost::shared_ptr<Mantid::Kernel::PropertyManager> > in PropertyManagerProperty.cpp.o
      vtable for Mantid::Kernel::PropertyManagerProperty in PropertyManagerProperty.cpp.o
  "Mantid::Kernel::PropertyWithValue<Mantid::Kernel::Matrix<double> >::saveProperty(NeXus::File*)", referenced from:
      vtable for Mantid::Kernel::MatrixProperty<double> in MatrixProperty.cpp.o
      vtable for Mantid::Kernel::PropertyWithValue<Mantid::Kernel::Matrix<double> > in MatrixProperty.cpp.o
  "Mantid::Kernel::PropertyWithValue<Mantid::Kernel::Matrix<float> >::saveProperty(NeXus::File*)", referenced from:
      vtable for Mantid::Kernel::MatrixProperty<float> in MatrixProperty.cpp.o
      vtable for Mantid::Kernel::PropertyWithValue<Mantid::Kernel::Matrix<float> > in MatrixProperty.cpp.o
  "Mantid::Kernel::PropertyWithValue<Mantid::Kernel::Matrix<int> >::saveProperty(NeXus::File*)", referenced from:
      vtable for Mantid::Kernel::MatrixProperty<int> in MatrixProperty.cpp.o
      vtable for Mantid::Kernel::PropertyWithValue<Mantid::Kernel::Matrix<int> > in MatrixProperty.cpp.o
  "Mantid::Kernel::PropertyWithValue<std::__1::vector<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >, std::__1::allocator<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > > > >::saveProperty(NeXus::File*)", referenced from:
      vtable for Mantid::Kernel::ArrayProperty<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > > in ArrayProperty.cpp.o
      vtable for Mantid::Kernel::PropertyWithValue<std::__1::vector<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >, std::__1::allocator<std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > > > > in ArrayProperty.cpp.o
  "Mantid::Kernel::PropertyWithValue<std::__1::vector<float, std::__1::allocator<float> > >::saveProperty(NeXus::File*)", referenced from:
      vtable for Mantid::Kernel::ArrayProperty<float> in PropertyNexus.cpp.o
      vtable for Mantid::Kernel::PropertyWithValue<std::__1::vector<float, std::__1::allocator<float> > > in PropertyNexus.cpp.o
  "Mantid::Kernel::PropertyWithValue<std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > >::saveProperty(NeXus::File*)", referenced from:
      vtable for Mantid::Kernel::ArrayProperty<unsigned long> in ArrayProperty.cpp.o
      vtable for Mantid::Kernel::PropertyWithValue<std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > > in ArrayProperty.cpp.o
  "Mantid::Kernel::PropertyWithValue<float>::saveProperty(NeXus::File*)", referenced from:
      vtable for Mantid::Kernel::PropertyWithValue<float> in PropertyNexus.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```